realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -1618,6 +1648,68 @@ public void copyToRealmOrUpdate_iterableChildObjects() {
     }
 
     @Test
+    public void copyToRealmOrUpdate_objectInOtherThreadThrows() {
```

Comment:
Actually I spend 1 hour trying to refactor the common stuff out, but the resulting code was much more convoluted and unreadable since the differences are subtle but still have a huge impact, so I went back to this approach.